lib/puma/single.rb

```diff
@@ -52,6 +52,8 @@ def run
       @server = server = start_server
       server_thread = server.run
 
+      @launcher.config.run_hooks :on_booted, nil, @launcher.events
```

Comment:
Rather than add this line, should we somehow make  use the already-existing `on_booted` event, I wonder? It's maybe not great that we have this `run_hooks` system which has the exact same name of a trigger as an `Event`... feels like they could be combined.

Reply:
Ok, that makes sense. I am not familiar yet how the events are working but I'll dig into that tomorrow and try to provide alternative solution that is addressing your suggestions. Thanks for this extremely quick feedback

Reply:
It's also possible that we decide it's possible, but we won't do it in this PR and leave the refactor for someone else later. But if you can do it on your own/it's easy enough, let's do it!